File: 0456-132-pattern/0456-132-pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
        stack<int> st;
        int nk = -0x3f3f3f3f;
        for(int i = nums.size()-1; i >= 0; i--){
            if(nums[i] < nk)
                return true;
            else{
                while(!st.empty() && st.top() < nums[i]){
                    nk = st.top();
                    st.pop();
                }
            }
            st.push(nums[i]);
        }
        
        return false;
    }
};

File: 0456-132-pattern/0456-132-pattern.py
class Solution:
    def find132pattern(self, nums: List[int]) -> bool:
        arr = []
        tmp = -0x3f3f3f3f

        for i in range(len(nums)-1, -1, -1):
            if nums[i] < tmp:
                return True
            else:
                while arr and arr[-1] < nums[i]:
                    tmp = arr[-1]
                    arr.pop()
            
            arr.append(nums[i])
        

        return False
        